internal/linker/linker.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2022, The Garble Authors.
+// See LICENSE for licensing information.
+
+package linker
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+const MagicValueEnv = "GARBLE_LNK_MAGIC"
+
+type overlayFile struct {
+	Replace map[string]string
+}
+
+func compileLinker(workingDirectory string, overlay map[string]string, outputLinkPath string) error {
+	file, _ := json.Marshal(&overlayFile{Replace: overlay})
```

Comment:
as a general rule, don't ignore errors :) if you think it should never happen, make it a panic - we have some of those already in the codebase. or just return it, as you already have an error return parameter.